src/v/storage/mvlog/readable_segment.cc

```diff
@@ -0,0 +1,20 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "storage/mvlog/readable_segment.h"
+
+#include "storage/mvlog/segment_reader.h"
+
+namespace storage::experimental::mvlog {
+
+std::unique_ptr<segment_reader> readable_segment::make_reader() {
+    return std::make_unique<segment_reader>(this);
```

Comment:
Out of curiosity, what sort of lifetime management measures/tools do you see us requiring between `readable_segment`s and `segment_reader`s in the future?

Reply:
My hope is that the code structure will make it trivially verifiable that the lifetimes are what they should be. Haven't given much thought to additional tooling though